tests/migration/paused.go

```diff
@@ -0,0 +1,190 @@
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			if vmi.Status.MigrationState != nil {
+				return vmi.Status.MigrationState.Mode
+			}
+			return v1.MigrationPreCopy
+		}, timeout, 1*time.Second).Should(Equal(expectedMode), fmt.Sprintf("migration should be in %s after %d s", expectedMode, timeout))
+		return vmi
+	}
+
+	Describe("Starting a VirtualMachineInstance ", func() {
+
+		Context("paused vm during migration", func() {
+
+			var migrationPolicy *migrationsv1.MigrationPolicy
+
+			BeforeEach(func() {
+				By("enable AllowWorkloadDisruption and limit migration bandwidth")
+				policyName := fmt.Sprintf("testpolicy-%s", rand.String(5))
+				migrationPolicy = kubecli.NewMinimalMigrationPolicy(policyName)
+				migrationPolicy.Spec.AllowWorkloadDisruption = kvpointer.P(true)
+				migrationPolicy.Spec.AllowPostCopy = kvpointer.P(false)
+				migrationPolicy.Spec.CompletionTimeoutPerGiB = kvpointer.P(int64(20))
+				migrationPolicy.Spec.BandwidthPerMigration = kvpointer.P(resource.MustParse("5Mi"))
+			})
+
+			Context("should migrate paused", func() {
+
+				applyMigrationPolicy := func(vmi *v1.VirtualMachineInstance) {
+					AlignPolicyAndVmi(vmi, migrationPolicy)
+					migrationPolicy = CreateMigrationPolicy(virtClient, migrationPolicy)
+				}
+
+				applyKubevirtCR := func() {
+					config := getCurrentKvConfig(virtClient)
+					config.MigrationConfiguration.AllowPostCopy = migrationPolicy.Spec.AllowPostCopy
+					config.MigrationConfiguration.AllowWorkloadDisruption = migrationPolicy.Spec.AllowWorkloadDisruption
+					config.MigrationConfiguration.CompletionTimeoutPerGiB = migrationPolicy.Spec.CompletionTimeoutPerGiB
+					config.MigrationConfiguration.BandwidthPerMigration = migrationPolicy.Spec.BandwidthPerMigration
+					tests.UpdateKubeVirtConfigValueAndWait(config)
+				}
+
+				type applySettingsType string
+				const (
+					applyWithMigrationPolicy applySettingsType = "policy"
+					applyWithKubevirtCR      applySettingsType = "kubevirt"
+					expectSuccess                              = true
+					expectFailure                              = false
+				)
+
+				Context("when acceptiable time exceeded and post-copy is forbidden", Serial, func() {
+					createLargeVirtualMachine := func() *v1.VirtualMachine {
+						vmi := libvmifact.NewFedora(
+							libnet.WithMasqueradeNetworking(),
+							libvmi.WithResourceMemory("3Gi"),
+							libvmi.WithRng())
+						vm := libvmi.NewVirtualMachine(vmi)
+
+						vm, err := virtClient.VirtualMachine(testsuite.NamespacePrivileged).Create(context.Background(), vm, metav1.CreateOptions{})
+						Expect(err).ToNot(HaveOccurred())
+						return vm
+					}
+
+					DescribeTable("should pause the VMI and ", func(expectSuccess bool, bandwidth string, settingsType applySettingsType) {
+
+						By("creating a large VM with RunStrategyRerunOnFailure")
```

Comment:
Why RerunOnFailure?

Reply:
Why not?

Reply:
Just kidding. This is not needed, I'll remove it. Thanks.